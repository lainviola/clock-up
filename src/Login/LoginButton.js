// @flow
import React, { Component } from 'react'
import Auth0Lock from 'auth0-lock'
import { withRouter } from 'react-router-dom'
import { AUTH0_ID_TOKEN } from '../globalConst'
import { Button } from '../components/Button'

type Props = {
  lock: Auth0Lock,
  // withRouter()
  match: any,
  location: any,
  history: any
}

export class LoginButton extends Component<Props> {
  showLogin: Function

  constructor(props: Props) {
    super(props)
    this.showLogin = this.showLogin.bind(this)
  }

  componentDidMount() {
    const { lock } = this.props
    // Auth0のログインモーダルで認証 -> コールバックURLへ帰還した時の処理
    lock.on('authenticated', authResult => {
      window.localStorage.setItem(AUTH0_ID_TOKEN, authResult.idToken)
      this.props.history.push(`/createuser`)
    })
  }

  /**
   * Auth0のログインモーダルを表示
   */
  showLogin() {
    const { lock } = this.props
    lock.show({
      auth: {
        params: {
          responseType: 'id_token token'
        }
      }
    })
  }

  render() {
    return (
      <MyButton onClick={this.showLogin} data-test="login-btn">
        Login
      </MyButton>
    )
  }
}

const MyButton = Button.extend`
  width: 200px;
  height: 50px;
  color: #66bb6a;
  border: 2px solid #66bb6a;
`

export default withRouter(LoginButton)
