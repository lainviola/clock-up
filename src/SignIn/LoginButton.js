// @flow
import React, { Component, PropTypes } from 'react'
import Auth0Lock from 'auth0-lock'
import { withRouter } from 'react-router-dom'
import { AUTH0_ID_TOKEN } from '../GlobalConst'

type Props = {
  // withRouter()
  match: any,
  location: any,
  history: any
}

class LoginButton extends Component<Props> {
  lock: Auth0Lock

  constructor(props) {
    super(props)

    // TODO constructor injection & add test
    const clientId = process.env.REACT_APP_AUTH0_CLIENT_ID
    const domain = process.env.REACT_APP_AUTH0_DOMAIN
    this.lock = new Auth0Lock(clientId, domain)
  }

  componentDidMount() {
    // Auth0のログインモーダルで認証 -> コールバックURLへ帰還した時の処理
    this.lock.on('authenticated', authResult => {
      window.localStorage.setItem(AUTH0_ID_TOKEN, authResult.idToken)
      this.props.history.push(`/signup`)
    })
  }

  /**
   * Auth0のログインモーダルを表示
   */
  showLogin = () => {
    this.lock.show({
      auth: {
        params: {
          responseType: 'id_token token'
        }
      }
    })
  }

  render() {
    return (
      <div>
        <span onClick={this.showLogin} data-test="sign-in-btn">Google Login</span>
      </div>
    )
  }
}

export default withRouter(LoginButton)
