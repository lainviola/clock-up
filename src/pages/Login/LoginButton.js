// @flow
import React, { Component } from 'react'
import { pure, compose } from 'recompose'
import Auth0Lock from 'auth0-lock'
import { withRouter } from 'react-router-dom'
import { AUTH0_ID_TOKEN } from '../../dataTypes'
import { Button } from '../../elements/Button'
import { theme } from '../../theme'
import type { WithRouterProps } from './../../propTypes'

const StyledButton = Button.extend`
  width: 200px;
  height: 50px;
`

type Props = {
  ...WithRouterProps,
  lock: Auth0Lock
}

export class LoginButton extends Component<Props> {
  showAuth0LoginModal: Function

  constructor(props: Props) {
    super(props)
    this.showAuth0LoginModal = this.showAuth0LoginModal.bind(this)
  }

  componentDidMount() {
    const { lock } = this.props
    // success Auth0 modal authentication -> come back to callback url from auth0
    lock.on('authenticated', authResult => {
      window.localStorage.setItem(AUTH0_ID_TOKEN, authResult.idToken)
      this.props.history.push(`/createuser`)
    })
  }

  showAuth0LoginModal() {
    const { lock } = this.props
    lock.show({
      auth: {
        params: {
          responseType: 'id_token token'
        }
      }
    })
  }

  render() {
    return (
      <StyledButton
        color={theme.green}
        onClick={this.showAuth0LoginModal}
        enzyme-testid="login-btn"
      >
        Login
      </StyledButton>
    )
  }
}

export default compose(
  withRouter,
  pure
)(LoginButton)
