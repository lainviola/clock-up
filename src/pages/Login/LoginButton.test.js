import React from 'react'
import { shallow } from 'enzyme'

import { LoginButton } from './LoginButton'
import Auth0Lock from 'auth0-lock'

describe('<LoginButton />', () => {
  function setup() {
    const lock = new Auth0Lock('story', 'book')
    const wrapper = shallow(<LoginButton lock={lock} />)
    return wrapper
  }

  it('should match to snapshot', () => {
    const wrapper = setup()
    expect(wrapper).toMatchSnapshot()
  })

  it('sould authenticated eventlistener is set of Auth0 SDK Object', () => {
    const wrapper = setup()
    const events = wrapper.instance().props.lock._events
    expect(Object.prototype.hasOwnProperty.call(events, 'authenticated')).toBe(
      true
    )
    expect(typeof events.authenticated).toEqual('function')
  })
})
