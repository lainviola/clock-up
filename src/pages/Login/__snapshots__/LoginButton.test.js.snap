// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> should match to snapshot 1`] = `
<styled.button
  color="#66bb6a"
  enzyme-testid="login-btn"
  onClick={[Function]}
>
  Login
</styled.button>
`;
